Documentation/planning/arcade-powered-source-build/in-arcade.md

```diff
@@ -0,0 +1,132 @@
+   original source code.
+
+The main doc discusses why this needs to be done in Arcade. We should also aim
+at all of this being doable via a single `./build.sh [...]` command. This allows
+source-build to be easily reproducible with a local build, something the project
+has been sorely missing and bars all but the most committed and awesome
+contributors.
+
+## Additions to the `build.sh` procedure
+
+It seems that source-build can fit into the existing Arcade extension points.
+
+### Tool Restore
+
+This point is normally used to restore tooling that may impact the ordinary
+restore by adding MSBuild targets. Source-build can add a step to get
+source-build dependencies, with several options:
+
+1. Use transport package. Restores the [intermediate nupkg] and sets it up as a
+   restore source.
+2. Bring-your-own dependencies. A set of args allow the user (or automation) to
+   point to existing intermediates on disk.
+3. Build dependencies from source. Instead of downloading intermediates, clone
+   the repos and build them all.
+
+Any of those options allows the source-build of the current repo to continue
+afterwards with source-built intermediates. Then, source-build sets up
+RestoreSources, a Package Version Props, a blob root, and any repo-specific
+special requirements.
+
+At this point in the build, we should also have all we need to create a
+source-buildable tarball. We should aim to allow the user to create it as early
+as possible and cancel the rest of the build, as they may be a distro maintainer
+who doesn't actually want to build the repo right now. Unfortunately, there are
+some known issues blocking us from creating a source-buildable tarball without
+building the product first, tracked by
+[source-build#831](https://github.com/dotnet/source-build/issues/831). Until we
+find solutions, tarball creation must be done in Publish instead.
+
+### Restore
+
+We need to be careful that no machine-wide caches are used here. Caching a
+source-built nupkg may cause unexpected and incredibly hard to detect errors
+later on.
+
+### Build/Pack
+
+The slew of source-build args normally passed to the build command should be
+moved to Arcade as much as possible. Many of these apply to the Build and Pack
+parts of the build process.
+
+### Publish
+
+This seems like a reasonable place to put [intermediate nupkg] creation. Not all
+repos use this target so care is needed to make sure this doesn't crash.
+
+Creating an intermediate nupkg involves gathering together all shipped artifacts
+(nupkgs and tarballs) into a nupkg. The nupkg the needs to be uploaded to the
+build pipeline along with a manifest to indicate to Arcade how to publish it.
+
+We should also ensure here that no prebuilts were used. This is a validation
+step, however it's essential for source-build health, and quick to check. The
+reason to do this during Publish is that there is no Test phase during a
+source-build and we don't know for sure whether any extra prebuilts were
```

Comment:
I don't think trying to get source-build-compatible tests is the best way to go... it seems like it would be better to focus on having tests able to run using the source-build artifacts even if the test infra uses non-source-buildable tooling or test data.

Either way, I agree it's worth mentioning in these docs.